src/app/features/collectibles/components/bitcoin/inscription.tsx

```diff
@@ -21,6 +48,14 @@ export function Inscription({ rawInscription }: InscriptionProps) {
   const navigate = useNavigate();
   const location = useLocation();
 
+  const limiter = useHiroApiRateLimiter();
+
+  useOnMount(() => {
+    if (isInscriptionContentFetchedFromHiroApi(inscription)) {
+      void limiter?.removeTokens(1);
+    }
+  });
```

Comment:
Any other ways to detect request without using onMount?

Reply:
CollectibleLoader?